gpt4all-chat/modellist.cpp

```diff
@@ -1502,9 +1503,94 @@ void ModelList::parseModelsJsonFile(const QByteArray &jsonData, bool save)
             { ModelList::ParametersRole, "?" },
             { ModelList::QuantRole, "NA" },
             { ModelList::TypeRole, "GPT" },
+            { ModelList::UrlRole, "https://api.openai.com/v1/chat/completions"},
         };
         updateData(id, data);
+        
     }
+    
+	const QString mistralDesc = tr("<ul><li>Requires personal Mistral API key.</li><li>WARNING: Will send"
+                                   " your chats to Mistral!</li><li>Your API key will be stored on disk</li><li>Will only be used"
+                                   " to communicate with Mistral</li><li>You can apply for an API key"
+                                   " <a href=\"https://console.mistral.ai/user/api-keys\">here.</a></li>");
+
+    {
+        const QString modelName = "Mistral Tiny API";
+        const QString id = modelName;
+        const QString modelFilename = "mistral-tiny.txt";
+        if (contains(modelFilename))
+            changeId(modelFilename, id);
+        if (!contains(id))
+            addModel(id);
+        QVector<QPair<int, QVariant>> data {
+            { ModelList::NameRole, modelName },
+            { ModelList::FilenameRole, modelFilename },
+            { ModelList::FilesizeRole, "minimal" },
+            { ModelList::OnlineRole, true },
+            { ModelList::DescriptionRole,
+             tr("<strong>Mistral Tiny model</strong><br>") + mistralDesc },
+            { ModelList::RequiresVersionRole, "2.4.2" },
+            { ModelList::OrderRole, "cb" },
```

Comment:
This key should be unique. E.g., `cc`, `cd`, `ce`

Reply:
fixed

Reply:
@Olyxz16 Using da, db, dc will also require updating models3.json:
https://github.com/nomic-ai/gpt4all/blob/0072860d24384cfb9197a0b88ba5e8dc2784ba03/gpt4all-chat/metadata/models3.json#L51

Reply:
fixed using cc, cd and ce